community/kernel/src/main/java/org/neo4j/kernel/impl/factory/GraphDatabaseFacade.java

```diff
@@ -650,11 +719,103 @@ private InternalTransaction beginTransactionInternal( KernelTransaction.Type typ
             }
         }
 
-        return getNodesByLabelAndPropertyWithoutIndex( propertyId, value, statement, labelId );
+        return getNodesByLabelAndPropertyWithoutIndex( statement, labelId, query );
+    }
+
+    private ResourceIterator<Node> nodesByLabelAndProperties( KernelTransaction transaction, int labelId, IndexQuery... queries )
+    {
+        Statement statement = transaction.acquireStatement();
+        Read read = transaction.dataRead();
+
+        if ( isInvalidQuery( labelId, queries ) )
+        {
+            statement.close();
+            return emptyResourceIterator();
+        }
+
+        int[] propertyIds = getSortedPropertyIds( queries );
```

Comment:
I just realized, before doing the whole sorted properties dance, we should make a quick check to see if there is an index the the properties in the order they were provided. That would likely match in many real situations, and should be much faster. 

If no index matches for the given query order, we can fall back to the sorted dance.